packages/trie/src/trie.ts

```diff
@@ -32,6 +32,16 @@ export class Trie<Data> {
     return node.data || (node.data = this.makeData(slice.call(array)));
   }
 
+  public peek<T extends any[]>(...array: T): Data | undefined {
+    return this.peekArray(array);
+  }
+
+  public peekArray<T extends IArguments | any[]>(array: T): Data | undefined {
+    let node: Trie<Data> | undefined = this;
+    forEach.call(array, (key) => (node = node && node.peekChildTrie(key)));
+    return node && node.data;
```

Comment:
```suggestion
  public peekArray<T extends IArguments | any[]>(array: T): Data | undefined {
    let node: Trie<Data> | undefined = this;
    for (let i = 0, len = array.length; node && i < len; ++i) {
      const map = this.weakness && isObjRef(array[i]) ? this.weak : this.strong;
      // If node ends up undefined here, the for loop will terminate immediately
      // because of the node && i < len condition, and return undefined below.
      node = map && map.get(array[i]);
    }
    return node && node.data;
```
In addition to the reasons you cited for adding `peek` and `peekArray`, there's another important reason this data structure benefits from a peeking operation: **you don't actually have to traverse the whole array to find out if `array` is present in the trie**, because you can bail out as soon as any prefix of the array does not appear to be present in the trie.

Instead of `Array.prototype.forEach`, a basic `for`-loop is more effective, though you could probably get `Array.prototype.some` or `.every` to provide the early termination benefit (not a recommendation, just a remark).

Also, given `peekChildTrie` is `private` and only used here, I don't see much benefit to decomposing it as a separate method.